doc/api/http.md

```diff
@@ -48,26 +48,39 @@ list like the following:
 added: v0.3.4
 -->
 
-The HTTP Agent is used for pooling sockets used in HTTP client
-requests.
-
-The HTTP Agent also defaults client requests to using
-`Connection: keep-alive`. If no pending HTTP requests are waiting on a
-socket to become free the socket is closed. This means that Node.js's
-pool has the benefit of keep-alive when under load but still does not
-require developers to manually close the HTTP clients using
-KeepAlive.
-
-If you opt into using HTTP KeepAlive, you can create an Agent object
-with that flag set to `true`.  (See the [constructor options][].)
-Then, the Agent will keep unused sockets in a pool for later use.  They
-will be explicitly marked so as to not keep the Node.js process running.
-However, it is still a good idea to explicitly [`destroy()`][] KeepAlive
-agents when they are no longer in use, so that the Sockets will be shut
-down.
-
-Sockets are removed from the agent's pool when the socket emits either
-a `'close'` event or a special `'agentRemove'` event. This means that if
+An `Agent` is responsible for managing connection persistence
+and reuse for HTTP clients. It maintains a queue of pending requests
+for a given host and port, reusing a single socket connection for each
+until the queue is empty, at which time the socket is destroyed.
+When a socket connection is no longer writable, for example, if a server
+closes the connection, the socket is destroyed.
+
+Unless an `Agent` is expressly provided in the [`http.request()`]
+options, all HTTP client requests will use the default
+[`http.globalAgent`].
+
+In addition to managing connection persistence for queued HTTP
```

Comment:
Addendum:

I would welcome some recommended wording here, because I have a hard time figuring out how to describe this behavior without leaking internals. The two concepts that you noted are correct. But by simply saying "when there are lots of requests" it's not clear what that means or how lots of _simultaneous_ requests would be created. I admit it's not clear in either case. Ugh.